cli/src/main.rs

```diff
@@ -106,12 +109,103 @@ fn run() -> Result<(), CliError> {
                             Arg::with_name("url")
                                 .short("U")
                                 .takes_value(true)
+                                .default_value(DEFAULT_SPLINTER_NODE_URL)
                                 .help("URL of node"),
                         ),
                 ),
         );
     }
 
+    #[cfg(feature = "node-registry")]
+    {
+        use action::nodes;
+        use clap::{Arg, SubCommand};
+
+        app = app.subcommand(
+            SubCommand::with_name("nodes")
+                .about("Interact with a node registry")
+                .arg(
+                    Arg::with_name(nodes::URL_ARG)
+                        .long("url")
+                        .short("U")
+                        .global(true)
+                        .takes_value(true)
+                        .default_value(DEFAULT_SPLINTER_NODE_URL)
+                        .help("Splinter node's endpoint"),
+                )
+                .subcommand(
+                    SubCommand::with_name("list")
+                        .about("List all nodes in the registry")
+                        .alias("ls")
+                        .arg(
+                            Arg::with_name(nodes::FORMAT_ARG)
+                                .long("format")
+                                .takes_value(true)
+                                .default_value(nodes::YAML)
+                                .possible_values(nodes::SUPPORTED_FORMATS)
+                                .help("Format nodes will be displayed in"),
+                        ),
+                )
+                .subcommand(
+                    SubCommand::with_name("get")
+                        .about("Get a single node from the registry")
+                        .arg(
+                            Arg::with_name("identity")
+                                .takes_value(true)
+                                .required(true)
+                                .help("Identity of the node to retrieve"),
+                        )
+                        .arg(
+                            Arg::with_name(nodes::FORMAT_ARG)
+                                .long("format")
+                                .takes_value(true)
+                                .default_value(nodes::YAML)
+                                .possible_values(nodes::SUPPORTED_FORMATS)
+                                .help("Format nodes will be displayed in"),
+                        ),
+                )
+                .subcommand(
+                    SubCommand::with_name("add")
```

Comment:
Could we get away from the file-based approach here? Or, better, support both using a file ("import") and a arg-based approach "add". If we added "import" we should also be able to "export".

Reply:
What would the arg-based "add" command look like? I'm not sure how useful that would be; it seems like it would be pretty clunky because of the number of arguments required (ID, endpoint, display name, metadata). Maybe @rbuysse and @desktophero would have some input on this.

Reply:
"add" from the command line would get messy, quick. Honestly, a preferred option from me would be to have an `apply` verb in the CLI that accepts an argument as a file path. 

If the object does not exist, `apply` based on the file.
If the object does exist, `apply` based on the file